### ISSUE TITLE
Deal with lock file changes polluting commits

### ISSUE THREAD

**Issue by humphd** [humphd/next#133]
Every time we `npm install` it updates `package-lock.json` and `npm-shrinkwrap.json` files throughout the tree.  This is annoying, and people not familiar with how these files work will often mess it up, rebases and merges get tricky due to merge conflicts, etc.

Let's see if we can improve this.  We could try [disabling them](https://github.com/substack/tape/commit/df48bfae19d8ba4b48055dacac8b81912b8887f2).  Since we do `npm install` in various places, we'll need to figure out how to pass that down, or have it get read from our root dir.

**Comment by humphd** [comment]
We could also look at adding these to our `npm install` invocations:

> The `--no-shrinkwrap` argument, which will ignore an available package lock or shrinkwrap file and use the package.json instead.
>
> The `--no-package-lock` argument will prevent npm from creating a package-lock.json file. When running with package-lock's disabled npm will not automatically prune your node modules when installing.

From https://docs.npmjs.com/cli/install

**Comment by dsych** [comment]
Is that a good idea? It seems like `package-lock.json` is a way for use to insure that everyone is running the same deps tree as well as speed-up `npm install` by skipping repeated deps. [see](https://stackoverflow.com/questions/44206782/do-i-commit-the-package-lock-json-file-created-by-npm-5)

**Comment by humphd** [comment]
@dsych agree.  So help me figure out a solution. Right now if you type `npm install` you end up with half-a-dozen `package-lock.json` and `npm-shrinkwrap.json` files being changed in `src/docs`, `src/editor`, etc.  This is really unexpected, especially for a person new to the project.  I'd like the workflow to be more natural than having to `git reset` or always `git commit` those changes in addition to your actual work.

**Comment by dsych** [comment]
As far as I remember, if you stage the lock files, git is going to discard
them as unmodified.
To your point about novice contributors, this workflow is not really
specific 5o our project, rather just a side effect of using npm and git.

**Comment by humphd** [comment]
Agree it's not specific to us; however, the fact that we do multiple invocations of `npm install` in child directories is very odd, and is why this ends up happening.

Probably the right move is to move some things out to their own repos and include them like we are with filer and v86, so you only have one lock to contend with instead of so many.
